homeassistant/components/broadlink/switch.py

```diff
@@ -54,12 +68,55 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
     mac_addr = config.get(CONF_MAC)
     switches = config.get(CONF_SWITCHES)
 
+    if not isinstance(switches, list):
+        switches = [
+            {
+                CONF_NAME: switch[CONF_FRIENDLY_NAME],
+                CONF_COMMAND_OFF: switch[CONF_COMMAND_OFF],
+                CONF_COMMAND_ON: switch[CONF_COMMAND_ON],
+            }
+            for switch in switches.values()
+        ]
+
+        _LOGGER.warning(
+            "Your configuration for the switch platform is deprecated. "
+            "Please refer to the Broadlink documentation to catch up"
+        )
+
     if mac_addr and switches:
+        for attr in {CONF_TYPE, CONF_TIMEOUT}:
+            if config.get(attr) is not None:
+                _LOGGER.warning(
+                    "The '%s' configuration variable is deprecated. "
+                    "Please remove it from your configuration file",
+                    attr,
+                )
+
         platform_data = hass.data[DOMAIN].platforms.setdefault(SWITCH_DOMAIN, {})
         platform_data.setdefault(mac_addr, []).extend(switches)
 
-    if host:
-        import_device(hass, host)
+        if host:
+            if import_device(hass, host):
+                _LOGGER.error(
```

Comment:
Do we really need this? We save the config yaml for the switch platform in hass.data and launch the switch platform set up again after setting up the config entry, by forwarding the entry. So when does this message come in to play?

Reply:
This message appears when we import the configuration of an RM switch but the device is not yet configured via the config flow. Here we don't have the entry yet, so we guide the user to create it.

If the entry already exists, we ask the user to remove the host option from the configuration, so we avoid this logic in the next startups.

Reply:
But we now import the configuration and set up the entry automatically, so then we don't need to show this message. There's no error?

Reply:
We should already have logic in place to not set up the same host more than once, right? We can still inform the user that the config yaml has been imported but that should not be an error message.

Reply:
> But we now import the configuration and set up the entry automatically, so then we don't need to show this message. There's no error?

Import is not automatic. The user needs to go to the integrations page, click Configure on the device, enter a name and select an area. If the device needs a reset we will guide the user through it, and if it is locked we will offer an unlock.

Reply:
Ok, but that is not an error. We're planning to have better default notifications in the frontend when an existing config entry needs more user input. Until then, we can show a persistent notification if needed. The log message should be max warning level.

Reply:
You are right. I'll make this better.